### PR TITLE
Add ability to customize facebook return field

### PR DESCRIPTION
**Pull request by XPLOT1ON**
This commit now allows the caller to provide facebook return field See [Facebook field](https://developers.facebook.com/docs/graph-api/reference/user/). And the returned response from success callback now has an additional key-value called other. It holds other fields from the original response that are not being used by the profile or access token key.

### CODE REVIEW

File: src/sdk/facebook.js [hunk 1]
```diff
@@ -7,31 +7,37 @@ let facebookScopes = [
...
+]
+
 /**
  * Loads Facebook SDK.
  * @param {string} appId
  * @param {array|string} scope
  * @see https://developers.facebook.com/docs/javascript/quickstart
  */
-const load = ({ appId, scope }) => new Promise((resolve) => {
+const load = ({ appId, scope, field }) => new Promise((resolve) => {
```
Comment (nicolas-goudry) [inline src/sdk/facebook.js]: Missing in JSDoc

File: src/sdk/facebook.js [hunk 2]
```diff
@@ -7,31 +7,37 @@ let facebookScopes = [
...
-  if (Array.isArray(scope)) {
-    facebookScopes = facebookScopes.concat(scope)
-  } else if (typeof scope === 'string' && scope) {
-    facebookScopes = facebookScopes.concat(scope.split(','))
-  }
-
-  facebookScopes = facebookScopes.reduce((acc, item) => {
-    if (typeof item === 'string' && acc.indexOf(item) === -1) {
-      acc.push(item.trim())
+  handleInput(scope).then(s => {
```
Comment (nicolas-goudry) [inline src/sdk/facebook.js]: Ok but why a promise ?

File: src/sdk/facebook.js [hunk 3]
```diff
@@ -56,6 +62,24 @@ const load = ({ appId, scope }) => new Promise((resolve) => {
...
+  items = items.reduce((acc, item) => {
+    if (typeof item === 'string' && acc.indexOf(item) === -1) {
+      acc.push(item.trim())
+    }
+    return acc
+  }, []).join(',')
+
+  return resolve(items)
+})
+
```
Comment (nicolas-goudry) [inline src/sdk/facebook.js]: I would do it this way:

```js
const handleInput = (input, defaultInput) => {
  let result

  if (Array.isArray(input)) {
    result = defaultInput.concat(input)
  } else if (input && typeof input === 'string') {
    result = defaultInput.concat(input.split(','))
  }

  result = result.reduce((acc, item) => {
    if (typeof item === 'string' && acc.indexOf(item) === -1) {
      acc.add(item.trim())
    }

    return acc
  }, new Set())

  return [...result]
}
```

And call would be `facebookFields = handleInput(field, facebookFields)`

File: src/sdk/facebook.js [hunk 4]
```diff
@@ -125,28 +149,37 @@ const logout = () => new Promise((resolve) => {
...
+      expiresAt: timestampFromNow(response.expiresIn)
+    },
+    other: Object.keys(response).reduce((ret, key) => {
+      if (excludeKeys.indexOf(key) === -1) {
+        ret[key] = response[key]
+      }
+      return ret
+    }, {})
+  })
+}
```
Comment (nicolas-goudry) [inline src/sdk/facebook.js]: I would do it this way:

```js
const generateUser = ({ accessToken, id, email, expiresIn, first_name, last_name, name, picture, userID, ...other }) => ({
  profile: {
    id,
    email,
    name,
    firstName: first_name,
    lastName: last_name,
    profilePicURL: picture.data.url
  },
  token: {
    accessToken,
    expiresAt: timestampFromNow(expiresIn)
  },
  other
})
```

**Comment by XPLOT1ON** [comment]
Yes, some of the code can be accomplished in a simpler way. I have made changes to reflect your suggestion.

**Pull request merged**